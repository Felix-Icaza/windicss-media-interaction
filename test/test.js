import { Processor } from 'windicss/lib'
import mediaInteractionVariants from '../src/index'

describe('Touch variants', () => {
  it('generate media query variants', () => {
    const processor = new Processor({
      variants: {
        backgroundColor: [
          'any-pointer-coarse',
          'any-pointer-fine',
          'any-pointer-none',
          'pointer-coarse',
          'pointer-fine',
          'pointer-none',
          'any-hover',
          'any-hover-none',
          'hover-hover',
          'hover-none'
        ],
        opacity: [
          'any-pointer-coarse',
          'any-pointer-fine',
          'any-pointer-none',
          'pointer-coarse',
          'pointer-fine',
          'pointer-none',
          'any-hover',
          'any-hover-none',
          'hover-hover',
          'hover-none'
        ]
      },
      plugins: [mediaInteractionVariants]
    })

    const css = `
      any-pointer-coarse:bg-blue-50
      any-pointer-fine:bg-blue-100
      any-pointer-none:bg-blue-200
      pointer-coarse:bg-blue-300
      pointer-fine:bg-blue-400
      pointer-none:bg-blue-500
      any-hover:bg-blue-600
      any-hover-none:bg-blue-700
      hover-hover:bg-blue-800
      hover-none:bg-blue-900

      any-pointer-coarse:opacity-0
      any-pointer-fine:opacity-10
      any-pointer-none:opacity-20
      pointer-coarse:opacity-30
      pointer-fine:opacity-40
      pointer-none:opacity-50
      any-hover:opacity-60
      any-hover-none:opacity-70
      hover-hover:opacity-80
      hover-none:opacity-90
    `

    const result = processor.interpret(css)

    expect(result.ignored.length).toEqual(0)
    expect(result.styleSheet.build()).toMatchSnapshot('media query variants')
  })

  it('extending media query variants', () => {
    const processor = new Processor({
      variants: {
        extend: {
          backgroundColor: [
            'any-pointer-coarse',
            'any-pointer-fine',
            'any-pointer-none',
            'pointer-coarse',
            'pointer-fine',
            'pointer-none',
            'any-hover',
            'any-hover-none',
            'hover-hover',
            'hover-none'
          ],
          opacity: [
            'any-pointer-coarse',
            'any-pointer-fine',
            'any-pointer-none',
            'pointer-coarse',
            'pointer-fine',
            'pointer-none',
            'any-hover',
            'any-hover-none',
            'hover-hover',
            'hover-none'
          ]
        }
      },
      plugins: [mediaInteractionVariants]
    })

    const css = `
      any-pointer-coarse:bg-blue-50
      any-pointer-fine:bg-blue-100
      any-pointer-none:bg-blue-200
      pointer-coarse:bg-blue-300
      pointer-fine:bg-blue-400
      pointer-none:bg-blue-500
      any-hover:bg-blue-600
      any-hover-none:bg-blue-700
      hover-hover:bg-blue-800
      hover-none:bg-blue-900

      any-pointer-coarse:opacity-0
      any-pointer-fine:opacity-10
      any-pointer-none:opacity-20
      pointer-coarse:opacity-30
      pointer-fine:opacity-40
      pointer-none:opacity-50
      any-hover:opacity-60
      any-hover-none:opacity-70
      hover-hover:opacity-80
      hover-none:opacity-90
    `

    const result = processor.interpret(css)

    expect(result.ignored.length).toEqual(0)
    expect(result.styleSheet.build()).toMatchSnapshot('extend variants')
  })

  it('generate group media query variants correctly', () => {
    const processor = new Processor({
      variants: {
        textColor: ['group-any-hover', 'group-hover-hover']
      },
      plugins: [mediaInteractionVariants]
    })

    const css = `
      group-any-coarse:bg-blue-50
      group-any-fine:bg-blue-100
      group-any-pnone:bg-blue-200
      group-pointer-coarse:bg-blue-300
      group-pointer-fine:bg-blue-400
      group-pointer-none:bg-blue-500
      group-any-hover:text-blue-600
      group-any-hnone:text-blue-700
      group-hover-hover:text-blue-800
      group-hover-none:text-blue-900
    `

    const result = processor.interpret(css)

    expect(result.ignored.length).toEqual(0)
    expect(result.styleSheet.build()).toMatchSnapshot(
      'group media query variants'
    )
  })
})
