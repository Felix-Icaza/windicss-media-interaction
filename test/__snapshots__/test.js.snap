// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touch variants extending media query variants: extend variants 1`] = `
"@media (any-hover: hover) {
  .any-hover\\\\:bg-blue-600:hover {
    --tw-bg-opacity: 1;
    background-color: rgba(37, 99, 235, var(--tw-bg-opacity));
  }
  .any-hover\\\\:opacity-60:hover {
    opacity: 0.6;
  }
}
@media (any-hover: none) {
  .any-hover-none\\\\:bg-blue-700 {
    --tw-bg-opacity: 1;
    background-color: rgba(29, 78, 216, var(--tw-bg-opacity));
  }
  .any-hover-none\\\\:opacity-70 {
    opacity: 0.7;
  }
}
@media (any-pointer: coarse) {
  .any-pointer-coarse\\\\:bg-blue-50 {
    --tw-bg-opacity: 1;
    background-color: rgba(239, 246, 255, var(--tw-bg-opacity));
  }
  .any-pointer-coarse\\\\:opacity-0 {
    opacity: 0;
  }
}
@media (any-pointer: fine) {
  .any-pointer-fine\\\\:bg-blue-100 {
    --tw-bg-opacity: 1;
    background-color: rgba(219, 234, 254, var(--tw-bg-opacity));
  }
  .any-pointer-fine\\\\:opacity-10 {
    opacity: 0.1;
  }
}
@media (any-pointer: none) {
  .any-pointer-none\\\\:bg-blue-200 {
    --tw-bg-opacity: 1;
    background-color: rgba(191, 219, 254, var(--tw-bg-opacity));
  }
  .any-pointer-none\\\\:opacity-20 {
    opacity: 0.2;
  }
}
@media (hover: hover) {
  .hover-hover\\\\:bg-blue-800:hover {
    --tw-bg-opacity: 1;
    background-color: rgba(30, 64, 175, var(--tw-bg-opacity));
  }
  .hover-hover\\\\:opacity-80:hover {
    opacity: 0.8;
  }
}
@media (hover: none) {
  .hover-none\\\\:bg-blue-900 {
    --tw-bg-opacity: 1;
    background-color: rgba(30, 58, 138, var(--tw-bg-opacity));
  }
  .hover-none\\\\:opacity-90 {
    opacity: 0.9;
  }
}
@media (pointer: coarse) {
  .pointer-coarse\\\\:bg-blue-300 {
    --tw-bg-opacity: 1;
    background-color: rgba(147, 197, 253, var(--tw-bg-opacity));
  }
  .pointer-coarse\\\\:opacity-30 {
    opacity: 0.3;
  }
}
@media (pointer: fine) {
  .pointer-fine\\\\:bg-blue-400 {
    --tw-bg-opacity: 1;
    background-color: rgba(96, 165, 250, var(--tw-bg-opacity));
  }
  .pointer-fine\\\\:opacity-40 {
    opacity: 0.4;
  }
}
@media (pointer: none) {
  .pointer-none\\\\:bg-blue-500 {
    --tw-bg-opacity: 1;
    background-color: rgba(59, 130, 246, var(--tw-bg-opacity));
  }
  .pointer-none\\\\:opacity-50 {
    opacity: 0.5;
  }
}"
`;

exports[`Touch variants generate group media query variants correctly: group media query variants 1`] = `
"@media (any-hover: hover) {
  .group:hover .group-any-hover\\\\:text-blue-600 {
    --tw-text-opacity: 1;
    color: rgba(37, 99, 235, var(--tw-text-opacity));
  }
}
@media (any-hover: none) {
  .group .group-any-hnone\\\\:text-blue-700 {
    --tw-text-opacity: 1;
    color: rgba(29, 78, 216, var(--tw-text-opacity));
  }
}
@media (any-pointer: coarse) {
  .group .group-any-coarse\\\\:bg-blue-50 {
    --tw-bg-opacity: 1;
    background-color: rgba(239, 246, 255, var(--tw-bg-opacity));
  }
}
@media (any-pointer: fine) {
  .group .group-any-fine\\\\:bg-blue-100 {
    --tw-bg-opacity: 1;
    background-color: rgba(219, 234, 254, var(--tw-bg-opacity));
  }
}
@media (any-pointer: none) {
  .group .group-any-pnone\\\\:bg-blue-200 {
    --tw-bg-opacity: 1;
    background-color: rgba(191, 219, 254, var(--tw-bg-opacity));
  }
}
@media (hover: hover) {
  .group:hover .group-hover-hover\\\\:text-blue-800 {
    --tw-text-opacity: 1;
    color: rgba(30, 64, 175, var(--tw-text-opacity));
  }
}
@media (hover: none) {
  .group .group-hover-none\\\\:text-blue-900 {
    --tw-text-opacity: 1;
    color: rgba(30, 58, 138, var(--tw-text-opacity));
  }
}
@media (pointer: coarse) {
  .group .group-pointer-coarse\\\\:bg-blue-300 {
    --tw-bg-opacity: 1;
    background-color: rgba(147, 197, 253, var(--tw-bg-opacity));
  }
}
@media (pointer: fine) {
  .group .group-pointer-fine\\\\:bg-blue-400 {
    --tw-bg-opacity: 1;
    background-color: rgba(96, 165, 250, var(--tw-bg-opacity));
  }
}
@media (pointer: none) {
  .group .group-pointer-none\\\\:bg-blue-500 {
    --tw-bg-opacity: 1;
    background-color: rgba(59, 130, 246, var(--tw-bg-opacity));
  }
}"
`;

exports[`Touch variants generate media query variants: media query variants 1`] = `
"@media (any-hover: hover) {
  .any-hover\\\\:bg-blue-600:hover {
    --tw-bg-opacity: 1;
    background-color: rgba(37, 99, 235, var(--tw-bg-opacity));
  }
  .any-hover\\\\:opacity-60:hover {
    opacity: 0.6;
  }
}
@media (any-hover: none) {
  .any-hover-none\\\\:bg-blue-700 {
    --tw-bg-opacity: 1;
    background-color: rgba(29, 78, 216, var(--tw-bg-opacity));
  }
  .any-hover-none\\\\:opacity-70 {
    opacity: 0.7;
  }
}
@media (any-pointer: coarse) {
  .any-pointer-coarse\\\\:bg-blue-50 {
    --tw-bg-opacity: 1;
    background-color: rgba(239, 246, 255, var(--tw-bg-opacity));
  }
  .any-pointer-coarse\\\\:opacity-0 {
    opacity: 0;
  }
}
@media (any-pointer: fine) {
  .any-pointer-fine\\\\:bg-blue-100 {
    --tw-bg-opacity: 1;
    background-color: rgba(219, 234, 254, var(--tw-bg-opacity));
  }
  .any-pointer-fine\\\\:opacity-10 {
    opacity: 0.1;
  }
}
@media (any-pointer: none) {
  .any-pointer-none\\\\:bg-blue-200 {
    --tw-bg-opacity: 1;
    background-color: rgba(191, 219, 254, var(--tw-bg-opacity));
  }
  .any-pointer-none\\\\:opacity-20 {
    opacity: 0.2;
  }
}
@media (hover: hover) {
  .hover-hover\\\\:bg-blue-800:hover {
    --tw-bg-opacity: 1;
    background-color: rgba(30, 64, 175, var(--tw-bg-opacity));
  }
  .hover-hover\\\\:opacity-80:hover {
    opacity: 0.8;
  }
}
@media (hover: none) {
  .hover-none\\\\:bg-blue-900 {
    --tw-bg-opacity: 1;
    background-color: rgba(30, 58, 138, var(--tw-bg-opacity));
  }
  .hover-none\\\\:opacity-90 {
    opacity: 0.9;
  }
}
@media (pointer: coarse) {
  .pointer-coarse\\\\:bg-blue-300 {
    --tw-bg-opacity: 1;
    background-color: rgba(147, 197, 253, var(--tw-bg-opacity));
  }
  .pointer-coarse\\\\:opacity-30 {
    opacity: 0.3;
  }
}
@media (pointer: fine) {
  .pointer-fine\\\\:bg-blue-400 {
    --tw-bg-opacity: 1;
    background-color: rgba(96, 165, 250, var(--tw-bg-opacity));
  }
  .pointer-fine\\\\:opacity-40 {
    opacity: 0.4;
  }
}
@media (pointer: none) {
  .pointer-none\\\\:bg-blue-500 {
    --tw-bg-opacity: 1;
    background-color: rgba(59, 130, 246, var(--tw-bg-opacity));
  }
  .pointer-none\\\\:opacity-50 {
    opacity: 0.5;
  }
}"
`;
